lensless/recon/trainable_recon.py

```diff
@@ -58,29 +65,56 @@ def __init__(self, psf, dtype=None, n_iter=1, **kwargs):
                 Data type to use for optimization.
             n_iter : int
                 Number of iterations for unrolled algorithm.
+            post_process : :py:class:`function` or :py:class:`~torch.nn.Module`, optional
+                If :py:class:`function` : Function to apply to the image estimate after the whole ADMM algorithm. Its input most be (image to process, noise_level), where noise_level is a learnable parameter. If it include aditional learnable parameters, they will not be added to the parameter list of the algorithm. To allow for traning, the function must be autograd compatible.
+                If :py:class:`~torch.nn.Module` : A DruNet like network. The network will be included as a submodule of the algorithm and its parameters will be added to the parameter list of the algorithm. If this isn't intended behavior, set requires_grad=False.
         """
         assert isinstance(psf, torch.Tensor), "PSF must be a torch.Tensor"
         super(TrainableReconstructionAlgorithm, self).__init__(
             psf, dtype=dtype, n_iter=n_iter, **kwargs
         )
+        if isinstance(post_process, torch.nn.Module):
+            # If the post_process is a torch module, we assume it is a DruNet like network.
+            from lensless.utils.image import process_with_DruNet
```

Comment:
is there something like an isinstance assert that you can do to make such it is a DruNet?

Reply:
It seems challenging since we don't want just to accept DruNet but any network with similar input and output. This is the best I could think of.